src/error.rs

```diff
@@ -104,6 +104,8 @@ pub enum Error {
     /// In response to a STATUS command, the server sent OK without actually sending any STATUS
     /// responses first.
     MissingStatusResponse,
+    /// Tag mismatch between client and server. New session must be created.
+    TagCorrupted(TagCorrupted),
```

Comment:
I think the name `TagCorrupted` is misleading. A "corrupted" tag feels more like if a tag isn't a number for example, or has somehow otherwise been modified in-flight. What this represents (as the doc string indicates) is a `TagMismatch` or `Desynchronized`, so let's call it one of those.